### PR TITLE
Add API method for computing transitive orderbook

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds a new API method for computing a transitive orderbook.

### Test Plan

```
$ cargo test -p pricegraph real_orderbooks -- --nocapture
    Finished test [unoptimized + debuginfo] target(s) in 0.10s
     Running target/debug/deps/pricegraph-397047f497bad3b3

running 1 test
#5298183: estimated order for buying 194.87633796952346 DAI for 1 WETH
#5298183: DAI-WETH market contains 91 ask orders and 100 bid orders within a 5% spread:
 - Ask orders
    ...
 - Buy orders
    ...
```

### CODE REVIEW

File: pricegraph/src/lib.rs [hunk 1]
```diff
@@ -134,6 +134,59 @@ impl Pricegraph {
             sell: sell_amount,
         })
     }
+
+    /// Computes a transitive orderbook for the given market specified by the
+    /// base and quote tokens.
+    ///
+    /// This method optionally accepts a spread is a decimal fraction that
```
Comment (e00E) [inline pricegraph/src/lib.rs:141]:
```suggestion
    /// This method optionally accepts a spread that is a decimal fraction that
```

File: pricegraph/src/lib.rs [hunk 2]
```diff
@@ -134,6 +134,59 @@ impl Pricegraph {
...
+                    sell: base,
+                },
+                spread,
+            ));
+
+        for orders in &mut [
+            &mut transitive_orderbook.asks,
+            &mut transitive_orderbook.bids,
+        ] {
+            orders.sort_unstable_by(|a, b| num::compare(a.price(), b.price()));
```
Comment (fleupold) [inline pricegraph/src/lib.rs:185]: If I understand correctly, the prices for bids and asks are going to be represented as inverse prices (thus the value will jump drastically between best bid/best ask). 
I believe normally bids and asks in an orderbook have the same price unit (e.g. https://poloniex.com/public?command=returnOrderBook&currencyPair=BTC_ETH&depth=50)

**Review by fleupold — APPROVED** [review]
Looks good. Just one thing to consider if the price unit should not unified for bids & asks.

File: pricegraph/src/lib.rs (hunk 2 above)
Comment (nlordell) [reply pricegraph/src/lib.rs:185]: One thing to note is that we return ask and bid orders, not prices. That is buy and sell amounts (as they are represented in the exchange).

I guess, one thing we can add is `inverse` method for inverting the orders and calculating the inverse price.

The sorting here is just so the ask orders are sorted, and the bid orders are sorted, and they are sorted interdependently.

Comment (nlordell) [reply pricegraph/src/lib.rs:185]: Maybe we can change the `TransitiveOrderbook` representation to align with expectations a bit more? We can also add iterator methods to produce `volume/price` pairs for the price estimator service?

Comment (nlordell) [reply pricegraph/src/lib.rs:185]: Created #1002  to capture this work.

**Review by e00E — APPROVED** [review]

**Pull request merged**